### PR TITLE
Added Memory Limits Not Defined query #492

### LINKED ISSUE #492: Add Memory Limits Not Defined query for Kubernetes
### Platform
Kubernetes

### Description
It's important to specify a memory limit.

### PR DESCRIPTION
**Pull request by joelcarvalhocheckmarx**
It's important to specify a memory limit.

Closes #492

### CODE REVIEW

File: assets/queries/k8s/memory_limits_not_defined/query.rego [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+
+CxPolicy [ result ] {
+   spec := input.document[i].spec 
+   limits := spec.containers[c].resources.limits
+   exists_memory := object.get(limits, "memory", "undefined") != "undefined"
+   not exists_memory
+   
+   result := {
+                "documentId": 		input.document[i].id,
+                "searchKey": 	    sprintf("spec.containers[%d].limits.memory", [c]),
```
Comment (joelsou5a) [inline assets/queries/k8s/memory_limits_not_defined/query.rego:11]: Add metadata filter:

"searchKey": sprintf("metadata.name=%s.spec.containers.name=%s.limits.memory", [metadata.name, containers[c].name]),

File: assets/queries/k8s/memory_limits_not_defined/query.rego [hunk 2]
```diff
@@ -0,0 +1,71 @@
+package Cx
+
+CxPolicy [ result ] {
+   spec := input.document[i].spec
```
Comment (joelsou5a) [inline assets/queries/k8s/memory_limits_not_defined/query.rego:4]: document := input.document[i]
spec := document .spec
metadata := document.metadata

File: assets/queries/k8s/memory_limits_not_defined/query.rego [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+
+
+CxPolicy [ result ] {
+   spec := input.document[i].spec
+   exists_containers := object.get(spec, "containers", "undefined") != "undefined"
+   not exists_containers
+   
+   result := {
+                "documentId": 		input.document[i].id,
+                "searchKey": 	    "spec",
```
Comment (joelsou5a) [inline assets/queries/k8s/memory_limits_not_defined/query.rego:26]: "searchKey": sprintf("metadata.name=%s.spec", [metadata.name]),

File: assets/queries/k8s/memory_limits_not_defined/query.rego [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+                "searchKey": 	    sprintf("spec.containers[%d].limits.memory", [c]),
+                "issueType":		   "MissingAttribute",
+                "keyExpectedValue": sprintf("spec.containers[%d].resources.limits.memory is defined", [c]),
+                "keyActualValue": 	sprintf("spec.containers[%d].resources.limits.memory is undefined", [c])
+              }
+}
+
+
+CxPolicy [ result ] {
+   spec := input.document[i].spec
```
Comment (joelsou5a) [inline assets/queries/k8s/memory_limits_not_defined/query.rego:20]: document := input.document[i]
spec := document.spec
metadata := document.metadata

File: assets/queries/k8s/memory_limits_not_defined/query.rego [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+   exists_containers := object.get(spec, "containers", "undefined") != "undefined"
+   exists_containers
+   
+   containers := spec.containers
+   exists_resources := object.get(containers[c], "resources", "undefined") != "undefined"
+   not exists_resources
+   
+   result := {
+                "documentId": 		input.document[i].id,
+                "searchKey": 	    sprintf("spec.containers[%d]", [c]),
```
Comment (joelsou5a) [inline assets/queries/k8s/memory_limits_not_defined/query.rego:44]: Add the changes do achieve this (like the ones on the comments before)

"searchKey": sprintf("metadata.name=%s.spec.containers.name=%s", [metadata.name, containers[c].name]),

File: assets/queries/k8s/memory_limits_not_defined/query.rego [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+   exists_resources := object.get(containers[c], "resources", "undefined") != "undefined"
+   exists_resources
+   
+   resources := spec.containers[c].resources
+   exists_limits := object.get(resources, "limits", "undefined") != "undefined"
+   not exists_limits
+   
+   result := {
+                "documentId": 		input.document[i].id,
+                "searchKey": 	    sprintf("spec.containers[%d]", [c]),
```
Comment (joelsou5a) [inline assets/queries/k8s/memory_limits_not_defined/query.rego:66]: Add the changes do achieve this (like the ones on the comments before)

"searchKey": sprintf("metadata.name=%s.spec.containers.name=%s", [metadata.name, containers[c].name]),

**Review by joelsou5a — CHANGES_REQUESTED** [review]

**Review by joaomartinscx — COMMENTED** [review]
Everything seems okay to me, I will recommend approval and merge of this PR.

**Review by ruigomescx — APPROVED** [review]

**Pull request merged**